vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -715,13 +744,36 @@ def _set_valid_labels(data, label):
             raise ValueError('Dataset must be pandas.DataFrame or d8.image_classification.Dataset')
 
 def _get_supported_models():
-    all_models = get_model_list()
-    blacklist = ['ssd', 'faster_rcnn', 'mask_rcnn', 'fcn', 'deeplab',
-                 'psp', 'icnet', 'fastscnn', 'danet', 'yolo', 'pose',
-                 'center_net', 'siamrpn', 'monodepth',
-                 'ucf101', 'kinetics', 'voc', 'coco', 'citys', 'mhpv1',
-                 'ade', 'hmdb51', 'sthsth', 'otb']
-    cls_models = [m for m in all_models if not any(x in m for x in blacklist)]
+    try:
+        import mxnet as _mxnet
+    except ImportError:
+        _mxnet = None
+    if _mxnet is not None:
+        from gluoncv.model_zoo import get_model_list
+        all_models = get_model_list()
+        blacklist = ['ssd', 'faster_rcnn', 'mask_rcnn', 'fcn', 'deeplab',
+                    'psp', 'icnet', 'fastscnn', 'danet', 'yolo', 'pose',
+                    'center_net', 'siamrpn', 'monodepth',
+                    'ucf101', 'kinetics', 'voc', 'coco', 'citys', 'mhpv1',
+                    'ade', 'hmdb51', 'sthsth', 'otb']
+        cls_models = [m for m in all_models if not any(x in m for x in blacklist)]
+    else:
+        cls_models = []
+    # add timm backend supported models
+    try:
+        import torch as _torch
+    except ImportError:
+        _torch = None
+    try:
+        import timm as _timm
+    except ImportError:
+        _timm = None
+    if _timm is not None:
+        cls_models += list(_timm.list_models())
+    elif _torch is None:
+        logger.warning('timm installed but torch is required to enable it.')
```

Comment:
This does not seem to trigger in the intended way. This never checks if TIMM was installed.

Reply:
timm-clean(no deps mock pure python package) is added to dependency so I assume timm is installed anyway, since import timm triggers import torch internally, the hack that will work is probably trying to catch the ImportError and parse it.